### PR TITLE
Beta-binomial

### PR DESCRIPTION
**Pull request by avi-stripe**
Adds the beta-binomial distribution.

### CODE REVIEW

File: rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala [hunk 1]
```diff
@@ -144,34 +144,40 @@ object Exponential {
...
-      (a - 1) *
-        u.log + (b - 1) *
-        (1 - u).log - Combinatorics.beta(a, b)
+  def binomial: Predictor[Int, Int, BetaBinomial] = Predictor.from { k: Int =>
+    BetaBinomial(a, b, k)
   }
 }
 
+object Beta {
+  def from(mean: Real, precision: Real) = Beta(mean * precision, (Real.one - mean) * precision)
```
Comment (avi-stripe) [inline rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala]: I'd love a more descriptive name for this method but `Beta.meanAndPrecision` seems too long and I can't think of anything better.

Comment (roban-stripe) [reply rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala]: `fromMoments` maybe (though I guess precision is the inverse of the second moment). or `fromSummaryStats`, `fromStats`?

Comment (avi-stripe) [reply rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala]: Hm, should we make it (mean,variance) instead and call it `Beta.moments` then?

Comment (roban-stripe) [reply rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala]: yeah, that's be fine with me

**Comment by roban-stripe** [comment]
LGTM, aside from any renaming you want to do on `Beta.from`.

**Comment by roban-stripe** [comment]
haha, why not both? great!

File: rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala [hunk 2]
```diff
@@ -175,7 +175,9 @@ final case class Beta(a: Real, b: Real) {
 }
 
 object Beta {
-  def from(mean: Real, precision: Real) = Beta(mean * precision, (Real.one - mean) * precision)
+  def meanAndPrecision(mean: Real, precision: Real) = Beta(mean * precision, (Real.one - mean) * precision)
+  def meanAndVariance(mean: Real, variance: Real) =
+    meanAndPrecision(mean, mean * (Real.one - mean) / variance - 1)
```
Comment (roban-stripe) [inline rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala]: Wait, @avi-stripe, isn't precision just 1/variance?

**Pull request merged**